compiler/noirc_frontend/src/hir/resolution/import.rs

```diff
@@ -309,7 +309,7 @@ fn resolve_external_dep(
     current_def_map: &CrateDefMap,
     directive: &ImportDirective,
     def_maps: &BTreeMap<CrateId, CrateDefMap>,
-    path_references: &mut Option<&mut Vec<ReferenceId>>,
+    path_references: &mut Option<&mut Vec<Option<ReferenceId>>>,
```

Comment:
Haha, sorry! 😅 

Maybe we can define a type alias for this? Actually in Rust Analyzer if you have something like this:

```rust
use some_crate::foo::bar;
```

when you hover over `some_crate` it shows the crate and its docs. But in this PR I skip the reference to the crate because just in this PR I realized that crates also show up in paths (in addition to modules). Or, well, I guess I was considering crates to be similar to modules.

Eventually we can let `some_crate` point to that crate (you'd jump to the `lib.rs` file, just like in Rust) and then all of a path's references will be `Some`, so we can remove that `Option`.

Reply:
Once this merges I'll capture an issue for this (though I won't work on it right now)

Reply:
No worries, I wasn't really being that serious. We can refactor this later.